pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,123 @@
+package localworkflows
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/content_type"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	FilterFindingsWorkflowName = "filterfindings"
+)
+
+var WORKFLOWID_FILTER_FINDINGS = workflow.NewWorkflowIdentifier(FilterFindingsWorkflowName)
+
+func InitFilterFindingsWorkflow(engine workflow.Engine) error {
+	flags := pflag.NewFlagSet(FilterFindingsWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_FILTER_FINDINGS, workflow.ConfigurationOptionsFromFlagset(flags), filterFindingsEntryPoint)
+
+	// engine.GetConfiguration().AddDefaultValue(configuration.FF_TRANSFORMATION_WORKFLOW, configuration.StandardDefaultValueFunction(false))
+	return err
+}
+
+type FindingsFilterFunc func(local_models.FindingResource) bool
+
+func contains(slice []string, item string) bool {
+	for _, s := range slice {
+		if s == item {
+			return true
+		}
+	}
+	return false
+}
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func filterFindingsBySeverity(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func filterFindings(findingsModel local_models.LocalFinding, filter []FindingsFilterFunc) local_models.LocalFinding {
+	filteredFindings := local_models.LocalFinding{}
+	for _, finding := range findingsModel.Findings {
+		match := true
+		for _, filter := range filter {
+			if match {
+				match = filter(finding)
+			}
+		}
+		if match {
+			filteredFindings.Findings = append(filteredFindings.Findings, finding)
+		}
+	}
+	return filteredFindings
+}
+
+func filterFindingsEntryPoint(invocationCtx workflow.InvocationContext, input []workflow.Data) (output []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	ff_transform_enabled := config.GetBool(configuration.FF_TRANSFORMATION_WORKFLOW)
+	output = []workflow.Data{}
+
+	logger.Println("Workflow Name:", FilterFindingsWorkflowName)
```

Comment:
question: will this set the appropriate log level? if not, consider using 

`logger.Debug().Msgf("Workflow Name: %s", FilterFindingsWorkflowName)`

Reply:
It will, Println sends a log event using debug level